### PR TITLE
Configure CA and update a few other projects' properties

### PR DESCRIPTION
**Pull request by drwill-ms**
CA = code analysis

https://learn.microsoft.com/en-us/visualstudio/code-quality/install-net-analyzers?view=vs-2022
https://learn.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2022

### CODE REVIEW

File: iothub/service/src/Microsoft.Azure.Devices.csproj [hunk 1]
```diff
@@ -2,10 +2,13 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <LangVersion>9.0</LangVersion>
+    <EnableNETAnalyzers>True</EnableNETAnalyzers>
+    <AnalysisLevel>latest-recommended</AnalysisLevel>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <RootNamespace>Microsoft.Azure.Devices</RootNamespace>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NeutralLanguage>en-US</NeutralLanguage>
```
Comment (patilsnr) [inline iothub/service/src/Microsoft.Azure.Devices.csproj:11]: Any reason why we're removing en-us from device and adding it to service? Not sure I understand

**Review by patilsnr — APPROVED** [review]

File: iothub/service/src/Microsoft.Azure.Devices.csproj (hunk 1 above)
Comment (drwill-ms) [reply iothub/service/src/Microsoft.Azure.Devices.csproj:11]: I didn't remove it from device. I moved where it was, so it shows up consistently in the same place in the csproj.

**Review by brycewang-microsoft — APPROVED** [review]

File: iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
     Warning for using pre-release nugets can be ignored since this sample 
     can't target a main release yet.
     -->
-    <NoWarn>NU5104</NoWarn>
+    <NoWarn>NU5104;CA2254</NoWarn>
```
Comment (abhipsaMisra) [inline iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.csproj:13]: document CA2254

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**